nms/server/middleware/__tests__/rateLimit-test.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import express, {
+  Request,
+  Response,
+  request as middlewareRequest,
+} from 'express';
+import {agent as request} from 'supertest';
+
+import {rateLimitMiddleware} from '..';
+
+const config = {
+  RATE_LIMIT_CONFIG: {
+    windowMs: 15 * 1000, // 15 seconds
+    limit: 2,
+  },
+};
+
+Object.defineProperty(global, "performance", {
+  writable: true,
+});
```

Comment:
Attempting this, pulled from https://github.com/facebook/react-native/issues/35701#issuecomment-1847579429